st2actions/st2actions/container/base.py

```diff
@@ -63,6 +67,10 @@ def _get_runner(self, runnertype_db, action_db, liveaction_db):
         runner.libs_dir_path = self._get_action_libs_abs_path(action_db.pack,
                                                               action_db.entry_point)
 
+        # For re-run, get the ActionExecutionDB in which the re-run is based on.
+        rerun_ref_id = runner.context.get('re-run', {}).get('ref')
+        runner.rerun_ex_ref = ActionExecution.get(id=rerun_ref_id) if rerun_ref_id else None
```

Comment:
Can we explicitly assign all the variables which get late / dynamically assigned to the runner as class level variables?

So:

```python
class RunnerContainer(object):
    rerun_ex_ref = None
    # etc.
```

Without that it's really hard to understand the code and know which variables are available when. I know it's not your fault and some of the code has been like that from the beginning, but we should try to improve it going forward when possible.